dev_process.md

```diff
@@ -0,0 +1,144 @@
+  versus a directed acyclic graph (DAG) with multiple commit
+  paths. Using `git merge` can create these multiple commit paths.
+
+* Issues do not have to be created for features as discussion can be
+  attached to the pull request. However, if you wish to discuss
+  design/implementations ideas beforehand, please feel free to create
+  an issue.
+
+We have a slack channel setup for GitHub notifications that are more
+extensive than the normal email notifications. Please contact the core
+team to be added.
+
+### Versioning and Release Process
+
+Versions and associated releases are driven by the recommendations and
+standards produced by the AIRR Community and their associated Working
+Groups, with specifications and software tools implementing those
+recommendations and standards. The various sub-projects within
+airr-standards do not have independent versioning but are
+interconnected together through the same version numbers. This way
+users know exactly which software versions are associated with which
+standards versions. This has the challenge that normal semantic
+version numbering (MAJOR.MINOR.PATCH) cannot be applied to the
+sub-projects, however the goal is that semantic versioning will apply
+to AIRR Standards as a whole. The design is thus:
+
+* MAJOR = MiAIRR version
+
+* MINOR = Implementation specifications
+
+* PATCH = Software updates
+
+New and updated software tools will increment the PATCH number,
+assuming that the implementation specifications have not changed. The
+current MiAIRR version is V1 (i.e. MAJOR = 1), and it is expected that
+this version will rarely change as it required formal adoption by the
+AIRR Community. Currently, MINOR = 0 as the implementation
+specifications are in pre-release form going through many
+changes. Once they are released (i.e. v1.1.0) then MINOR will be
+incremented whenever a backwards incompatible change is introduced.
+
+The airr-standards repository uses release branches to maintain stable
+releases separate from ongoing development. Each release branch is for
+a specific MAJOR.MINOR version, with tags being assigned on that
+branch for every new PATCH release. When development on `master` is
+deemed ready for release, a branch is created from `master` called
+release-MAJOR.MINOR where MAJOR.MINOR.0 will be the first associated
+release tag. The `master` branch will then be development for a future
+MAJOR.MINOR release, for example if the current release branch is
+release-1.1, then the `master` branch will be development for a future
+1.2.0 or 2.0.0 release. All work related to a prior MAJOR.MINOR
+release, which essentially should only be software updates and bug
+fixes, should be commited to that release branch. Note: a release may
+just tag `master` and not create the actual release branch until later
+when it is needed.
+
+### Code Style
+
+**General Guidelines**
+
+* Do not commit `.gitignore` files or IDE project files.
```

Comment:
Don't people typically comit `.gitignore`?

Reply:
I don't know about typically, but definitely often. No one voiced an opinion, so I just made a decision.

It's totally fine with me if we want to commit `.gitignore` and project files instead. We just need some simply rules about it.

Reply:
I usually commit `.gitignore`. Are there pros/cons?

Reply:
Committing `.gitignore` helps avoid user error (accidentally committing things that should be committed).

It can get a bit cluttered with everyone's individual environment specific stuff. Eg, IDE files, test run directories, etc. There's some potential for error if people use the same names for something that will be part of the repo in the future (eg, a `tests` folder), but that's very rare I would say.

I think committing `.gitignore` is more pro than con.

I think committing IDE project files is more con than pro though, because then it propagates environment settings and temp files. With the exception of Rstudio `.Rproj` files. Those are lightweight and contain build parameters.